src/Core/src/Platform/iOS/WindowExtensions.cs

```diff
@@ -96,6 +100,128 @@ public static void UpdateMinimumHeight(this UIWindow platformWindow, IWindow win
+		static Selector StandardWindowButtonSetHidden => StandardWindowButtonSetHiddenSelector ??= new Selector("setHidden:");
+		
+		enum NSWindowButton : ulong
+		{
+			MinimizeButton,
+			MaximizeButton
+		}
+
+		public static async Task UpdateIsMinimizableAsync(this UIWindow platformWindow, IWindow window)
+		{
+			var nsWindow = await GetNSWindowFromUIWindowAsync(platformWindow);
+
+			if (nsWindow != null)
+			{
+				// Get the minimize button from the window
+				var minimizeButton = Runtime.GetNSObject(
+					IntPtr_objc_msgSend_IntPtr(
+						nsWindow.Handle,
+						StandardWindowButton.Handle,
+						(IntPtr)(ulong)NSWindowButton.MinimizeButton));
+
+				if (minimizeButton is not null)
+				{
+					// Update the button's visibility based on the window's state
+					bool isMinimizable = window.IsMinimizable;
+					
+					void_objc_msgSend_bool(
+						minimizeButton.Handle,
+						StandardWindowButtonSetHidden.Handle,
+						!isMinimizable);
+				}
+			}
+		}
+		
+		public static async Task UpdateIsMaximizableAsync(this UIWindow platformWindow, IWindow window)
+		{
+			var nsWindow = await GetNSWindowFromUIWindowAsync(platformWindow);
+
+			if (nsWindow != null)
+			{
+				// Get the maximize button from the window
+				var maximizeButton = Runtime.GetNSObject(
+					IntPtr_objc_msgSend_IntPtr(
+						nsWindow.Handle,
+						StandardWindowButton.Handle,
+						(IntPtr)(ulong)NSWindowButton.MaximizeButton));
+
+				if (maximizeButton is not null)
+				{
+					// Update the button's visibility based on the window's state
+					bool isMaximizable = window.IsMaximizable;
+					
+					void_objc_msgSend_bool(
+						maximizeButton.Handle,
+						StandardWindowButtonSetHidden.Handle,
+						!isMaximizable);
+				}
+			}
+		}
+
+		static async Task<NSObject?> GetNSWindowFromUIWindowAsync(this UIWindow window)
+		{
+			if (window is null)
+			{
+				return null;
+			}
+
+			var nsApplication = Runtime.GetNSObject(Class.GetHandle("NSApplication"));
+			if (nsApplication is null)
+			{
+				return null;
+			}
+
+			var sharedApplication = nsApplication.PerformSelector(new Selector("sharedApplication"));
+			if (sharedApplication is null)
+			{
+				return null;
+			}
+
+			var applicationDelegate = sharedApplication.PerformSelector(new Selector("delegate"));
+			if (applicationDelegate is null)
+			{
+				return null;
+			}
+
+			return await GetNSWindowAsync(window, applicationDelegate);
+		}
+
+		static async Task<NSObject?> GetNSWindowAsync(UIWindow window, NSObject applicationDelegate)
+		{
+			// Custom selector used to bridge host window with UIKit UIWindow
+			var nsWindowHandle = IntPtr_objc_msgSend_IntPtr(applicationDelegate.Handle,
+				Selector.GetHandle("hostWindowForUIWindow:"), window.Handle);
+			var nsWindow = Runtime.GetNSObject<NSObject>(nsWindowHandle);
+			
+			if (nsWindow is null)
+			{
+				// Delay to allow async window resolution
+				await Task.Delay(500);
+				return await GetNSWindowAsync(window, applicationDelegate);
```

Comment:
Can this go into an infinite recursion? Maybe only try a certain number of times?